app/src/main/java/com/azuresamples/msalandroidapp/MSGraphRequestWrapper.java

```diff
@@ -24,71 +24,76 @@
-                                               @NonNull final Response.ErrorListener errorListener) {
-        Log.d(TAG, "Starting volley request to graph");
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    @SuppressWarnings("unchecked")
+    public static CompletableFuture<String> callGraphAPI(@NonNull final String graphResourceUrl,
+                                                         @NonNull final String accessToken) {
+        Log.d(TAG, "Starting SDK request to graph");
 
         /* Make sure we have a token to send to graph */
         if (accessToken == null || accessToken.length() == 0) {
-            return;
+            return CompletableFuture.completedFuture(null);
         }
 
-        RequestQueue queue = Volley.newRequestQueue(context);
-        JSONObject parameters = new JSONObject();
+        StaticTokenAuthProvider authProvider =
+            new StaticTokenAuthProvider(accessToken);
 
-        try {
-            parameters.put("key", "value");
-        } catch (Exception e) {
-            Log.d(TAG, "Failed to put parameters: " + e.toString());
-        }
+        GraphServiceClient graphClient = GraphServiceClient.builder()
+            .authenticationProvider(authProvider)
+            .buildClient();
 
-        JsonObjectRequest request = new JsonObjectRequest(Request.Method.GET, graphResourceUrl,
-                parameters, responseListener, errorListener) {
-            @Override
-            public Map<String, String> getHeaders() {
-                Map<String, String> headers = new HashMap<>();
-                headers.put("Authorization", "Bearer " + accessToken);
-                return headers;
-            }
-        };
+        String relativeUrl = "";
 
-        Log.d(TAG, "Adding HTTP GET to Queue, Request: " + request.toString());
+        if (graphResourceUrl.toLowerCase().contains(MS_GRAPH_V1_ENDPOINT)) {
+            relativeUrl = graphResourceUrl.substring(MS_GRAPH_V1_ENDPOINT.length());
+        } else if (graphResourceUrl.toLowerCase().contains(MS_GRAPH_BETA_ENDPOINT)) {
+            relativeUrl = graphResourceUrl.substring(MS_GRAPH_BETA_ENDPOINT.length());
+        }
 
-        request.setRetryPolicy(new DefaultRetryPolicy(
-                3000,
-                DefaultRetryPolicy.DEFAULT_MAX_RETRIES,
-                DefaultRetryPolicy.DEFAULT_BACKOFF_MULT));
-        queue.add(request);
+        // The customRequest builder allows you to call any relative URL
+        // (like "/me") and get back a generic object
+        // For an example that uses the SDK's fluent API to return strongly-typed objects,
+        // see https://github.com/microsoftgraph/msgraph-training-android
+        // or https://docs.microsoft.com/en-us/graph/tutorials/android
+        return graphClient.customRequest(relativeUrl)
+            .buildRequest()
+            .getAsync()
+            .thenApply(response -> graphClient.getSerializer().serializeObject(response));
```

Comment:
shouldn't this be deserialize object?

Reply:
No - the app displays the JSON string, and response is an `JsonObject`.

Reply:
response.body().string() is sufficient then :)

Reply:
Actually looks like just `response.toString()` does it. I wasn't sure because the compiler sees it as `Object`.